### ISSUE TITLE
dyld: Library not loaded: @rpath/libswiftCore.dylib   Referenced from: /Users/stathis/Library/Developer/Xcode/DerivedData/zzzzxxxdddd-gxpbgcpggospzxavpqtfdhnkuaez/Build/Products/Debug-iphonesimulator/WCLShineButton.framework/WCLShineButton   Reason: image not found (lldb) 

### ISSUE THREAD

**Issue by efstathiosntonas** [prscX/react-native-shine-button#7]
I followed the installation instructions but i get this error: 

```
dyld: Library not loaded: @rpath/libswiftCore.dylib
  Referenced from: /Users/stathis/Library/Developer/Xcode/DerivedData/christian_tete-gxpbgcpggospzxavpqtfdhnkuaez/Build/Products/Debug-iphonesimulator/WCLShineButton.framework/WCLShineButton
  Reason: image not found
(lldb) 

```

Xcode version 9.2 (9C40b), iOS 11.2, iPhone 8 simulator, iPhone 7 real device

**Comment by prscX** [comment]
Thanks @efstathiosntonas for raising the issue.

Have you added WCLShineButton.framework in your app Embedded Binaries? Please follow below steps in order to add the same:

Add WCLShineButton.framework in Embedded Binaries & Linked Frameworks and Libraries. You can find this library from below path:

`node_modules/react-native-shine-button/ios/WCLShineButton/WCLShineButton.xcodeproj`

<img src="https://github.com/prscX/react-native-shine-button/raw/master/assets/setup.png" />

Let me know in case it does not work.

Thanks
Pranav

**Comment by efstathiosntonas** [comment]
Yes, i followed installation instructions, any particular reason on the fainted framework toolbox on the left of the framework name?

![screen shot 2018-03-19 at 16 04 57](https://user-images.githubusercontent.com/717975/37600371-d10dbf44-2b8f-11e8-87a1-3760d1ab317b.png)

**Comment by prscX** [comment]
Hi @efstathiosntonas,

Are you able to link the library? Let me know in case you need any help.

Thanks
Pranav

**Comment by jorgemarrero** [comment]
I have the same problem, and I also tried with```Add Other```

**Comment by jorgemarrero** [comment]
What do you have in ```Build Setting/Header Search Paths```

**Comment by efstathiosntonas** [comment]
I cant tell since i’m not using it now.

**Comment by prscX** [comment]
Hi @efstathiosntonas + @jorgemarrero: Apologies for not able to provide any help.

Can you please try once by running the built in example app. Just wanted to check is it software stack related issue or something else.

Thanks
</ Pranav >

**Comment by yelkamel** [comment]
I got the same issue 
I tried everything.
any news about that ?

**Comment by jorgemarrero** [comment]
@prscX sorry, I haven't had time to investigate the problem. 

@yelkamel My temporary 'solution' was not to use this component in iOS, I only use it in Android where it works fine.

**Comment by JayricMok** [comment]
I having the same issue, any idea to fix it ?

**Comment by JayricMok** [comment]
I tried to updated to latest version 0.0.8 and follow the pod install step and set the framework to swift language 3.3, but when I compile, I get this error

**RNShineButton.m**
![image](https://user-images.githubusercontent.com/6957533/39166863-938985f4-47be-11e8-99a4-05fbe991e81d.png)

![image](https://user-images.githubusercontent.com/6957533/39166854-80bf8e64-47be-11e8-93eb-5198d9c21771.png)

@prscX , any idea to fix this ?

**Comment by prscX** [comment]
@JayricMok: Have you run below command in order to install native swift library?

`cd ../node_modules/react-native-shine-button/ios && pod install` 

If yes, can you please compare your local install file from github file:

- `../node_modules/react-native-shine-button/ios/Pods/WCLShineButton/WCLShineButton/WCLShineButton.swift`
- `https://github.com/prscX/WCLShineButton/blob/master/WCLShineButton/WCLShineButton.swift`

Ideally both should be same. If they are not same can you please run below command:

- `pod repo update`
- `pod install`

Thanks
</ Pranav >

**Comment by JayricMok** [comment]
@prscX yes, I do exactly your step, and stiil get those error, I'm using

react-native-cli: 2.0.1
react-native: 0.53.0

**Comment by JayricMok** [comment]
unfortunately i think i have no choice to remove react-native-shine-button from ios, i still have to continue the task, hope @prscX have any update or solution. Btw Android worked fine.

**Comment by prscX** [comment]
@JayricMok: That's strange. Just for confirmation, can you please run built-in example app in order to verify the same.

Thanks
</ Pranav >

**Comment by prscX** [comment]
@JayricMok: My apologies for that. Across packages those have native Swift library dependencies, I am receiving issues from the community.

Earlier I have tried to come up with a concrete solution but was unable to find the right one. I will again try this from different angle.

Thanks
</ Pranav >

**Comment by prscX** [comment]
Hi @JayricMok: I have got the root cause and solution. Basically `xcode` is caching previous version of library, you need to clear `Derived Data`. Please follow below few references in order to perform the same:

- https://stackoverflow.com/questions/38016143/how-can-i-delete-derived-data-in-xcode-8
- https://stackoverflow.com/questions/24039932/how-to-delete-derived-data-in-xcode-6-and-later
- https://iosdevcenters.blogspot.com/2015/12/how-to-delete-derived-data-and-clean.html

Please let me know in case it does not work.

Thanks
</ Pranav >
